lib/jekyll/tags/link.rb

```diff
@@ -0,0 +1,29 @@
+module Jekyll
+  module Tags
+    class Link < Liquid::Tag
+      TagName = 'link'
+
+      def initialize(tag_name, relative_path, tokens)
+        super
+
+        @relative_path = relative_path.strip
+      end
+
+      def render(context)
+        site = context.registers[:site]
+
+        site.docs_to_write.each do |document|
+          return document.url if document.relative_path == @relative_path
+        end
+
+        raise ArgumentError.new <<-eos
+Could not find document "#{@relative_path}" in tag '#{TagName}'.
```

Comment:
For removing the "new" yes.  For the former, it's up for debate @jekyll/core will have to decide that though as of late it seems a lot of errors are shifting that way because it's cleaner to read.